shopinvader_api_delivery_carrier/routers/delivery_carrier.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from typing import Annotated
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException
+
+from odoo import api, fields, models
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.delivery.models.delivery_carrier import DeliveryCarrier
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+
+from ..schemas import (
+    DeliveryCarrier as DeliveryCarrierSchema,
+    DeliveryCarrierSearch,
+    DeliveryCarrierWithPrice,
+)
+
+delivery_carrier_router = APIRouter(tags=["delivery_carriers"])
+
+
+@delivery_carrier_router.get("/delivery_carriers")
+def search(
+    data: Annotated[DeliveryCarrierSearch, Depends()],
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> list[DeliveryCarrierSchema]:
+    """
+    Returns the list of all available carriers.
+
+    The field type is a technical field only use inform if the carrier
+    provides some specialized functionalities
+    """
+    delivery_carriers = (
+        env["shopinvader_api_delivery_carrier.delivery_carrier_router.helper"]
+        .new({"partner": partner})
+        ._search(data, cart=None)
+    )
+    return [
+        DeliveryCarrierSchema.from_delivery_carrier(delivery_carrier)
+        for delivery_carrier in delivery_carriers
+    ]
+
+
+@delivery_carrier_router.get("/{uuid}/delivery_carriers")
+@delivery_carrier_router.get("/current/delivery_carriers")
```

Comment:
I keep seeing this `current`... what's the point of it? I know it comes from the cart endpoint but I don't get why we need a specific key to default to the last used cart.

Reply:
Because the default route `@delivery_carrier_router.get("/delivery_carriers")` is used to return all delivery carriers, not only the ones applicable to the current cart.